src/sas/sascalc/invariant/invariant.py

```diff
@@ -424,6 +424,25 @@ def __init__(self, data, background=0, scale=1):
         # Extrapolation range
         self._low_q_limit = Q_MINIMUM
 
+    @property
+    def background(self):
+        return self._background
+
+    @background.setter
+    def background(self, value):
+        self._background = value
+
+    @property
+    def scale(self):
+        return self._scale
+
+    @scale.setter
+    def scale(self, value):
+        self._scale = value
+
+    def set_data(self, data):
+        self._data = self._get_data(data)
+
```

Comment:
I find mixing explicit setters/getters (`set_data`, `get_data`) and implicit (through decorators) confusing, we should only have one consistent API for the given calculator.